Stripe/STPAPIClient.m

```diff
@@ -163,7 +166,12 @@ - (NSMutableURLRequest *)configuredRequestForURL:(NSURL *)url {
     NSMutableDictionary *additionalHeaders = [NSMutableDictionary new];
     additionalHeaders[@"X-Stripe-User-Agent"] = [self.class stripeUserAgentDetailsWithAppInfo:self.appInfo];
     additionalHeaders[@"Stripe-Version"] = APIVersion;
-    additionalHeaders[@"Authorization"] = [@"Bearer " stringByAppendingString:self.apiKey ?: @""];
+    NSString *authorizationBearer = self.apiKey ?: @"";
+    if (self.useEphemeralKeyForNextRequest && self.ephemeralKey.secret) {
+        self.useEphemeralKeyForNextRequest = NO;
```

Comment:
why is this reset after the first request?

Reply:
the desired behavior is that we use the ephemeral key only when hitting the Customer API. 

There are a bunch of different ways to implement this and I have no fondness for this one though.

Reply:
hmm, I don't like that this has the built-in assumption that this is coming from a short-lived instance. 

What do you think about adding `+ (NSURLSessionTask *)<action>WithAPIClient:endpoint:parameters:additionalRequestHeaders:(nullable NSDictionary *)...` methods?

Reply:
instance meaning the shared client configured to use an ephemeral key. Then we would call the new STPAPIRequest methods from the customer STPAPIClient methods

Reply:
Gave it a shot, I like this better.